I get the same error I get in the "4.1 upgrade" branch.

```
BUILD FAILED
/wrkdirs/usr/ports/databases/cassandra4/work/apache-cassandra-4.1.0-src/.build/build-resolver.xml:204: java.net.UnknownHostException: files.pythonhosted.org
```

As far as I can tell it's a different Maven repo (use by the sigar sub-project, I think) to connect to and cache…?

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 PORTNAME=	cassandra
-DISTVERSION=	4.0.5
+DISTVERSION=	4.0.8
 CATEGORIES=	databases java
 MASTER_SITES=	https://archive.apache.org/dist/${PORTNAME}/${DISTVERSION}/:apache \
 		https://repo1.maven.org/maven2/com/github/luben/zstd-jni/1.5.0-4/:maven

--- a/Makefile
+++ b/Makefile
@@ -7,16 +7,19 @@ PKGNAMESUFFIX=	4
 DISTNAME=	apache-${PORTNAME}-${DISTVERSION}-src
 DISTFILES=	${DISTNAME}.tar.gz:apache \
 		${ZSTD_DISTFILE} \
-		apache-${PORTNAME}-${DISTVERSION}-repo.tar.xz:repo
+		${MAVEN_CACHE_FILE}:repo
+DIST_SUBDIR=	${PORTNAME}
 EXTRACT_ONLY=	${DISTNAME}.tar.gz \
-		apache-${PORTNAME}-${DISTVERSION}-repo.tar.xz
+		${MAVEN_CACHE_FILE}
 
 MAINTAINER=	language.devel@gmail.com
 COMMENT=	Highly scalable distributed database
+WWW=		https://cassandra.apache.org/
 
 LICENSE=	APACHE20
 LICENSE_FILE=	${WRKSRC}/LICENSE.txt
 
+FETCH_DEPENDS=	ant:devel/apache-ant
 RUN_DEPENDS=	snappyjava>=0:archivers/snappy-java \
 		netty>0:java/netty
 
@@ -81,17 +84,27 @@ DOCS_BUILD_DEPENDS=	${PYTHON_PKGNAMEPREFIX}sphinx>=0,1:textproc/py-sphinx@${PY_F
 
 PORTDOCS=		*
 
+MAVEN_CACHE_FILE=	apache-${PORTNAME}-${DISTVERSION}-repo.tar.xz
+
+.if !exists(${DISTDIR}/${DIST_SUBDIR}/${MAVEN_CACHE_FILE})
 pre-fetch:
-	${MKDIR} -p ${WRKSRC}/.build
-	${MKDIR} -p ${WRKSRC}/src/java
+	${MKDIR} ${DISTDIR}/${DIST_SUBDIR}
+	${MKDIR} ${WRKSRC}/.build
+	${MKDIR} ${WRKSRC}/src/java
 	${CP} ${FILESDIR}/maven/build.* ${WRKSRC}
 	${CP} ${FILESDIR}/maven/build-* ${WRKSRC}/.build
 	cd ${WRKSRC} && ${ANT} -Dmaven.repo.local=${REPO_DIR} -Dlocal.repository=${REPO_DIR} ${USEJDK11} resolver-dist-lib
 	cd ${REPO_DIR} && ${FIND} . -type f -name "*.repositories" -a -exec ${SED} -i '' -e '2s,.*,Mon Aug 08 20:40:04 CEST 2022,' {} +
-	cd ${REPO_DIR} && ${FIND} . -print0 | xargs -0 touch -d 2022-08-08T20:40:04Z
-	cd ${WRKDIR} && ${FIND} -s repository/ -print >list.txt && \
-            ${TAR} cJf ${DISTDIR}/apache-${PORTNAME}-${DISTVERSION}-repo.tar.xz -nT list.txt
-	${RM} -r ${WRKDIR}
+	cd ${WRKDIR} && ${MTREE_CMD} -cbnSp repository | ${MTREE_CMD} -C | ${SED} \
+		-e 's:time=[0-9.]*:time=0.000000000:' \
+		-e 's:\([gu]id\)=[0-9]*:\1=0:g' \
+		-e 's:flags=.*:flags=none:' \
+		-e 's:^\.:./repository:' \
+		> maven-offline-cache.mtree
+	cd ${WRKDIR} && ${TAR} cJf ${DISTDIR}/${DIST_SUBDIR}/${MAVEN_CACHE_FILE} \
+		@maven-offline-cache.mtree
+	ls -al ${DISTDIR}/${DIST_SUBDIR}/apache-${PORTNAME}-${DISTVERSION}-repo.tar.xz
+.endif
 
 do-build:
 	@${DO_NADA} # Do nothing: Prevent USE_ANT from running a default build target.

--- a/Makefile
+++ b/Makefile
@@ -182,7 +182,7 @@ ZSTD_DISTFILE=
 post-install:
 	${LN} -s ${JAVAJARDIR}/netty.jar ${STAGEDIR}${DATADIR}/lib/netty.jar
 .if ${ARCH} == amd64 || ${ARCH} == i386
-	${CP} ${DISTDIR}/zstd-jni-${ZSTDJNI_VERSION}-freebsd_${ARCH}.jar ${STAGEDIR}${DATADIR}/lib/
+	${CP} ${DISTDIR}/${DIST_SUBDIR}/zstd-jni-${ZSTDJNI_VERSION}-freebsd_${ARCH}.jar ${STAGEDIR}${DATADIR}/lib/
 .endif
 
 post-install-DOCS-on:

--- a/distinfo
+++ b/distinfo
@@ -1,9 +1,7 @@
-TIMESTAMP = 1659987276
-SHA256 (apache-cassandra-4.0.5-src.tar.gz) = 9d79bcd8a9791436ded40b3e4d4c4b9d3bda77ab16d57059d1883e3ee99587aa
-SIZE (apache-cassandra-4.0.5-src.tar.gz) = 12872179
-SHA256 (zstd-jni-1.5.0-4-freebsd_amd64.jar) = b653460b6ff374db2c01f39a7b7cdb44008c9efd55dc96ae5f869abe9f58d180
-SIZE (zstd-jni-1.5.0-4-freebsd_amd64.jar) = 670993
-SHA256 (zstd-jni-1.5.0-4-freebsd_i386.jar) = 1bcb75c5837e42d10ec0193ba9df099874792f0829b0ce8dd3412303c6454595
-SIZE (zstd-jni-1.5.0-4-freebsd_i386.jar) = 604557
-SHA256 (apache-cassandra-4.0.5-repo.tar.gz) = f2c6dd937b953762f3a6aa1b7e04ead1ae636d376e5d08f703ce668a521e1835
-SIZE (apache-cassandra-4.0.5-repo.tar.gz) = 81273048
+TIMESTAMP = 1678485438
+SHA256 (cassandra/apache-cassandra-4.0.5-src.tar.gz) = 9d79bcd8a9791436ded40b3e4d4c4b9d3bda77ab16d57059d1883e3ee99587aa
+SIZE (cassandra/apache-cassandra-4.0.5-src.tar.gz) = 12872179
+SHA256 (cassandra/zstd-jni-1.5.0-4-freebsd_amd64.jar) = b653460b6ff374db2c01f39a7b7cdb44008c9efd55dc96ae5f869abe9f58d180
+SIZE (cassandra/zstd-jni-1.5.0-4-freebsd_amd64.jar) = 670993
+SHA256 (cassandra/apache-cassandra-4.0.5-repo.tar.xz) = e11d49fa71376bb31de041b54c51ec1ea077afe511d3b28b5479afcc41075e0f
+SIZE (cassandra/apache-cassandra-4.0.5-repo.tar.xz) = 78033936

--- a/distinfo
+++ b/distinfo
@@ -1,7 +1,7 @@
-TIMESTAMP = 1678485438
-SHA256 (cassandra/apache-cassandra-4.0.5-src.tar.gz) = 9d79bcd8a9791436ded40b3e4d4c4b9d3bda77ab16d57059d1883e3ee99587aa
-SIZE (cassandra/apache-cassandra-4.0.5-src.tar.gz) = 12872179
+TIMESTAMP = 1678487126
+SHA256 (cassandra/apache-cassandra-4.0.8-src.tar.gz) = 98da97d7fe850a0a24eb2aef212e982f73205cbd63955d2915b4ad02e28dae9c
+SIZE (cassandra/apache-cassandra-4.0.8-src.tar.gz) = 12953548
 SHA256 (cassandra/zstd-jni-1.5.0-4-freebsd_amd64.jar) = b653460b6ff374db2c01f39a7b7cdb44008c9efd55dc96ae5f869abe9f58d180
 SIZE (cassandra/zstd-jni-1.5.0-4-freebsd_amd64.jar) = 670993
-SHA256 (cassandra/apache-cassandra-4.0.5-repo.tar.xz) = e11d49fa71376bb31de041b54c51ec1ea077afe511d3b28b5479afcc41075e0f
-SIZE (cassandra/apache-cassandra-4.0.5-repo.tar.xz) = 78033936
+SHA256 (cassandra/apache-cassandra-4.0.8-repo.tar.xz) = c31de1ecb1acd6865b1e4485d4b9b3dbf86fec2a38e9e789bb3ade97d898ef16
+SIZE (cassandra/apache-cassandra-4.0.8-repo.tar.xz) = 78382652

--- a/files/maven/build.xml
+++ b/files/maven/build.xml
@@ -544,7 +544,7 @@
             <exclusion groupId="com.google.guava" artifactId="guava"/>
           </dependency>
           <dependency groupId="org.apache.cassandra" artifactId="dtest-api" version="0.0.13" scope="test"/>
-          <dependency groupId="org.reflections" artifactId="reflections" version="0.9.12" scope="test"/>
+          <dependency groupId="org.reflections" artifactId="reflections" version="0.10.2" scope="test"/>
           <dependency groupId="org.apache.hadoop" artifactId="hadoop-core" version="1.0.3" scope="provided">
             <exclusion groupId="org.mortbay.jetty" artifactId="servlet-api"/>
             <exclusion groupId="commons-logging" artifactId="commons-logging"/>

--- a/pkg-plist
+++ b/pkg-plist
@@ -95,6 +95,7 @@
 %%DATADIR%%/pylib/cqlshlib/util.py
 %%DATADIR%%/pylib/cqlshlib/wcwidth.py
 %%DATADIR%%/pylib/README.asc
+%%DATADIR%%/pylib/pytest.ini
 %%DATADIR%%/pylib/requirements.txt
 %%DATADIR%%/pylib/setup.py
 %%DATADIR%%/tools/bin/auditlogviewer

